src/avalanche.ts

```diff
@@ -46,7 +46,11 @@ export default class AvalancheCore {
     this.ip = ip;
     this.port = port;
     this.protocol = protocol;
-    this.url = `${protocol}://${ip}:${port}`;
```

Comment:
We can probably still use this old implementation instead of having the if by doing something like
```js
this.url = `${protocol}://${ip ?? ''}:${port}`;
```

Reply:
I believe this would work if it was `this.url = '${protocol}://${ip}${port ? ':${port}' : ''}';`

Reply:
Yup, that's another approach!